### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by IanVS**
Attempt to ensure:
- All anonymous functions have a space: `function (){}`
- All named functions have no space: `function myFunc(){}`
- All commas are followed by a space
- All JavaScript files begin with `'use strict';`
- All if statements followed by space: `if (true)`
- No duplicate definition of variables
- No trailing spaces
- No unused declarations or requires
- No missing or extra semicolons
- ES6 require syntax is used on requires

Conventions followed from Douglas Crockford: http://javascript.crockford.com/code.html

### CODE REVIEW

**Comment by kriswill** [comment]
The ES6 transpiler automatically puts in the "use strict" stuff.  Also, I'm not a fan of Crockford's style.  see: https://github.com/jshttp/style-guide.

File: lib/helpers/dockerCompose.js [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 require('shelljs/global');
 
-module.exports = function() {
-  if(which('docker-compose')) {
+module.exports = function () {
```
Comment (kriswill) [inline lib/helpers/dockerCompose.js]: not sure I like this, functions aren't cyclomatics.

File: lib/helpers/dockerCompose.js [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
+  if (which('docker-compose')) {
```
Comment (kriswill) [inline lib/helpers/dockerCompose.js]: I like this, `if` is a cyclomatic.  I think the rule should be cyclomatic keywords are surrounded by a space.

File: lib/tasks/getTemplates.js [hunk 3]
```diff
@@ -15,15 +15,15 @@ module.exports = function getTemplates(folderPath) {
   var walked = walkSync(path.join(folderPath, 'templates'));
 
   //Only leave files (plus their relative path) in the walked array
-  var walkedFiles = walked.filter(function(element){
+  var walkedFiles = walked.filter(function (element){
```
Comment (kriswill) [inline lib/tasks/getTemplates.js]: instead of `function (xx){` I would do it `function(xx) {`
space before the opening curly.

**Comment by IanVS** [comment]
I really have no dog in the style fight.  I'm new to JS, and have no strong opinions other than that code within a project should be consistent.  That was my only intent here, to start a dialog and try to take a step towards standardizing the code style.  Whatever @Globegitter @kriswill and @mgenev agree on is perfectly fine with me, and I will update the PR with the results.  Perhaps the only suggestion I have is to try to follow the same conventions as ember and/or ember-cli, since that would make it easier to get cross-pollination from ember devs.  That said, I don't know what their style is, I'm sure you all have more experience reading their source.  

TL;DR: just let me know how you want it and I'll clean up the PR and put together a short style guide we can post on the CONTRIBUTING.md

File: lib/tasks/getTemplates.js (hunk 3 above)
Comment (IanVS) [inline lib/tasks/getTemplates.js]: Good point, that's something I didn't check for and should have.

**Comment by kriswill** [comment]
@IanVS yeah, you're intention is good, I really do appreciate your contributions.  I'm sure all of us have our personal preferences, when it comes to style.  I will defer to Markus on this one.

**Comment by Globegitter** [comment]
Yep @kriswill I did try the colon-less for a while and actually there is one 'bug' about it in JS as far as I am aware. When you start a line with `[` I believe it was. But anyway, I do like colon-less and will actually go for that in my next project but want to leave this as-is for now.

And yes @IanVS as far as I can tell that is exactly ember code style and thanks again for this useul contribution, plus the style outlining for our future `CONTRIBUTION.md`.

**Comment by kriswill** [comment]
Yeah, the edge case for semicolon-free javascript is statements that start with `(` or `[`.  To me, it seems the edge case should be the exception, instead of forcing a rule on every other line of code.  So, I simply start lines that use a bracket or paren with `;(` or `;[`, problem solved.

**Comment by IanVS** [comment]
I learned programming mostly on python, and really liked the semantic whitespace and lack of silly things like curly braces and semicolons.  Also the style guidelines were very clear how to write "correct" python ([PEP-8](https://www.python.org/dev/peps/pep-0008/)).  I believe PHP is similar, having strong style recommendations (PSRs).  I looked for something similar in JS, but it seems the closest thing out there is the Crockford recommendations I linked to above.  Are there no "official" style guides for JavaScript?

**Comment by kriswill** [comment]
yeah, above I linked [JSHTTP style-guides](https://github.com/jshttp/style-guide), which is the default style guild for many of the node.js javascrtipt HTTP component projects here: https://jshttp.github.io, and various other projects that TJ Hallowaychuck has implemented.  It was popularized by the [Bootstrap](https://github.com/twbs/bootstrap/blob/master/js/transition.js) project.
